drivers/serial/uart_xlnx_ps.c

```diff
@@ -313,6 +385,446 @@ static void uart_xlnx_ps_poll_out(struct device *dev, unsigned char c)
 	} while ((reg_val & XUARTPS_SR_TXEMPTY) == 0);
 }
 
+/**
+ * @brief Converts a parity enum value to a Mode Register bit mask.
+ *
+ * Converts a value of an enumeration type provided by the driver
+ * framework for the configuration of the UART's parity setting
+ * into a bit mask within the Mode Register.
+ *
+ * It is assumed that the Mode Register contents that are being
+ * modified within this function come with the bits modified by
+ * this function already masked out by the caller.
+ *
+ * @param mode_reg Pointer to the Mode Register contents to which
+ *                 the parity configuration shall be added.
+ * @param parity   Enumeration value to be converted to a bit mask.
+ *
+ * @return Indication of success, always true for this function
+ *         as all parity modes supported by the API are also supported
+ *         by the hardware.
+ */
+static inline bool uart_xlnx_ps_cfg2ll_parity(
+	u32_t *mode_reg,
+	enum uart_config_parity parity)
+{
+	/*
+	 * Translate the new parity configuration to the mode register's
+	 * bits [5..3] (PAR):
+	 *  000b : even
+	 *  001b : odd
+	 *  010b : space
+	 *  011b : mark
+	 *  1xxb : none
+	 */
+
+	switch (parity) {
+	default:
+	case UART_CFG_PARITY_EVEN:
+		*mode_reg |= XUARTPS_MR_PARITY_EVEN;
+		break;
+	case UART_CFG_PARITY_ODD:
+		*mode_reg |= XUARTPS_MR_PARITY_ODD;
+		break;
+	case UART_CFG_PARITY_SPACE:
+		*mode_reg |= XUARTPS_MR_PARITY_SPACE;
+		break;
+	case UART_CFG_PARITY_MARK:
+		*mode_reg |= XUARTPS_MR_PARITY_MARK;
+		break;
+	case UART_CFG_PARITY_NONE:
+		*mode_reg |= XUARTPS_MR_PARITY_NONE;
+		break;
+	}
+
+	return true;
```

Comment:
In this and the following few functions return `true` or `false` for success or fail, respectively, Can these be `1` or `0` to be consistent with the rest of the file (mostly, some return enum)?

Reply:
As long as the function return type is `bool` (i.e. `stdbool.h`), it should use `true` and `false` as per the C standard.

Reply:
@stephanosio That's something to consider next UART API update, but I think `uart_irq_tx_ready()` should also be able to return `-ENOTSUP`.

Reply:
Agreed - I would like to see -ENOTSUP as well, but I won't hold this PR back for it. Thanks for the consideration!